### ISSUE TITLE
pytorch.profiler cannot profile aten:mm on GPU

### ISSUE THREAD

**Issue by scse-l** [pytorch/kineto#799]
I use pytorch.profiler to profile a program of matmul on GPU, it seems profiler does not record aten.mm correctly. There is stats in GPU kernel View,

<img width="2118" alt="image" src="https://github.com/pytorch/kineto/assets/11534916/dc126d48-1517-4af2-9200-8fd37aeaa6a4">

 but no GPU kernel stats in Trace view.

<img width="1903" alt="image" src="https://github.com/pytorch/kineto/assets/11534916/d4d747af-9b88-47b3-88eb-a3e3a9d00ef1">

Sample code:
```python
import torch
a = torch.rand([1, 1024, 2048], device='cuda')
b = torch.rand([2048, 2048], device='cuda')
with torch.profiler.profile(
    activities=[
        torch.profiler.ProfilerActivity.CPU,
        torch.profiler.ProfilerActivity.CUDA,
    ],
    on_trace_ready=torch.profiler.tensorboard_trace_handler("./mm-profile")
):
    torch.matmul(a, b)
```

**Comment by scse-l** [comment]
Overall view
<img width="2142" alt="image" src="https://github.com/pytorch/kineto/assets/11534916/421d2c76-5339-44d9-be33-9b8ff01788a6">

**Comment by scse-l** [comment]
Run `torch.matmul` for 100 times, there will be stats in Trace view.
It seems that there is a lot of warm-up work for aten:mm's first run. What does cudaFreee do?
<img width="2104" alt="image" src="https://github.com/pytorch/kineto/assets/11534916/d3eb30ca-dff7-4dac-91b5-95d1cce244e3">

**Comment by aaronenyeshi** [comment]
Hi, you can try turning on with_stack=True. You may get a little more detail on why its in cudaFree.

**Comment by scse-l** [comment]
> Hi, you can try turning on with_stack=True. You may get a little more detail on why its in cudaFree.

Sure, I'll try it. I appreciate your guidance.
